sycl/source/detail/plugin.hpp

```diff
@@ -236,7 +241,7 @@ class plugin {
   std::shared_ptr<std::mutex> getPluginMutex() { return MPluginMutex; }
 
 private:
-  RT::PiPlugin MPlugin;
+  std::shared_ptr<RT::PiPlugin> MPlugin;
```

Comment:
Shouldn't we have one to one mapping between `plugin` and `RT::PiPlugin` and one to many between `plugin` and `platform` ?

Reply:
You're right. Though, adding such a change to this patch will make it more complex for review.
Shifting to 1-1 plugin-pi_plugin and 1-n plugin-platform is the next step.

Reply:
Don't we already have 1:many mapping to platforms at line 253?

Reply:
@smaslov-intel , there's such a mapping between `sycl::detail::plugin` and `pi_platform`. Though, I have an intent to add the same mapping for `sycl::detail::plugin` and `sycl::detail::platform_impl`.